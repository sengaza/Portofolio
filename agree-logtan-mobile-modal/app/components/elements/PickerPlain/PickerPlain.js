import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';
import ModalDropdown from 'react-native-modal-dropdown';
import PropTypes from 'prop-types';
import styles from './PickerPlain.styles';
import { COLOR_BASE_TERTIARY_MAIN, COLOR_GREY_40 } from '../../../styles';
import { scale, verticalScale } from '../../../utils/scaling';
import SvgDropdown from '../../../../assets/svgs/Dropdown';
import I18n from '../../../i18n';

export default class PickerPlain extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false
    };
  }

  _dropdownShow = () => {
    this.setState({ open: true });
  };

  _dropdownHide = () => {
    this.setState({ open: false });
  };

  renderRow(option) {
    return (
      <TouchableOpacity style={{ height: 35, paddingLeft: 18, justifyContent: 'center' }}>
        <Text style={styles.textStyle2}>{option.name}</Text>
      </TouchableOpacity>
    );
  }

  renderSeparator() {
    return <View />;
  }

  render() {
    const {
      options,
      placeholder,
      selected,
      customPickerPlainStyle,
      customTextStyle,
      customDropdownStyle,
      customContainer
    } = this.props;
    // eslint-disable-next-line no-unused-vars
    const { open } = this.state;
    return (
      <View style={customContainer}>
        <ModalDropdown
          {...this.props}
          options={options}
          style={[styles.PickerPlainContainer, customPickerPlainStyle]}
          dropdownStyle={[
            {
              borderColor: COLOR_BASE_TERTIARY_MAIN,
              borderWidth: 1,
              height: (35 + StyleSheet.hairlineWidth) * (options.length < 6 ? options.length : 5),
              marginBottom: 5,
              marginLeft: scale(-14),
              marginTop: verticalScale(10),
              width: scale(295)
            },
            customDropdownStyle
          ]}
          dropdownTextStyle={styles.textStyle2}
          dropdownTextHighlightStyle={styles.textStyle}
          renderRow={this.renderRow}
          renderSeparator={this.renderSeparator}
          onDropdownWillShow={this._dropdownShow}
          onDropdownWillHide={this._dropdownHide}
        >
          <View
            style={{
              flexDirection: 'row',
              width: '100%',
              justifyContent: 'space-between',
              alignItems: 'center',
              borderBottomWidth: 1,
              borderColor: COLOR_GREY_40,
              paddingBottom: 5
            }}
          >
            {selected < 0 ? (
              <Text style={[styles.placeholderStyle, customTextStyle]}>{placeholder}</Text>
            ) : (
              <Text style={[styles.textStyle, customTextStyle]}>{options[selected].name}</Text>
            )}
            {/* <Image
              source={open ? IMAGES.dropdownClose : IMAGES.dropdownOpen}
              style={styles.image}
              resizeMode="contain"
            /> */}
            <View style={styles.image}>
              <SvgDropdown />
            </View>
          </View>
        </ModalDropdown>
      </View>
    );
  }
}

PickerPlain.propTypes = {
  options: PropTypes.array,
  selected: PropTypes.number,
  placeholder: PropTypes.string,
  customPickerPlainStyle: PropTypes.number,
  customDropdownStyle: PropTypes.number,
  customTextStyle: PropTypes.number,
  customContainer: PropTypes.object
};

PickerPlain.defaultProps = {
  options: [{ name: 'Option1', id: '12' }, { name: 'Option2', id: '11' }],
  selected: 12,
  placeholder: I18n.t('chooseItem'),
  customPickerPlainStyle: null,
  customTextStyle: null,
  customDropdownStyle: null,
  customContainer: null
};
