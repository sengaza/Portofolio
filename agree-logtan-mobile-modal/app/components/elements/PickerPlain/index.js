import PickerPlain from './PickerPlain';

export default PickerPlain;
