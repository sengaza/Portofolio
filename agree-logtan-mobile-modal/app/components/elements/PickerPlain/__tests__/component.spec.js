import { assertSnapshots, mockSetState } from '../../../../utils/TestUtils/snapshot';
import PickerPlain from '../PickerPlain';

jest.mock('../../../../configs', () => ({
  IMAGES: {}
}));

describe('Component snapshot testing', () => {
  const configs = [
    {
      props: { options: [0, 1, 2] },
      desc: 'renders without data'
    },
    {
      state: { open: true },
      props: { options: [0, 1, 2, 3, 4] },
      desc: 'renders without data'
    },
    {
      state: { open: true },
      props: { options: [0, 1, 2, 3, 4, 5] },
      desc: 'renders without data'
    }
  ];
  assertSnapshots(PickerPlain, configs);
});

describe('PickerPlain', () => {
  const ComponentWithMockedState = mockSetState(PickerPlain);
  // eslint-disable-next-line no-unused-vars
  const getInstance = props => new ComponentWithMockedState(props);
  beforeEach(() => {
    this.props = {
      navigation: {
        goBack: jest.fn()
      }
    };
  });

  afterEach(() => {
    this.props = null;
    jest.clearAllMocks();
  });

  test('_dropdownShow', async () => {
    const instance = getInstance(this.props);
    await instance._dropdownShow();
  });

  test('_dropdownHide', async () => {
    const instance = getInstance(this.props);
    await instance._dropdownHide();
  });

  test('renderRow', async () => {
    const option = '';
    const instance = getInstance(this.props);
    await instance.renderRow(option);
  });

  test('renderSeparator', async () => {
    const instance = getInstance(this.props);
    await instance.renderSeparator();
  });
});
