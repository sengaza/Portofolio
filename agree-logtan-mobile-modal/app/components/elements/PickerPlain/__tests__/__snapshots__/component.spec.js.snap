// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component snapshot testing renders without data 1`] = `
<View>
  <View
    animated={true}
    defaultIndex={-1}
    defaultValue="Please select..."
    disabled={false}
    dropdownStyle={
      Array [
        Object {
          "borderColor": "#bdbdbd",
          "borderWidth": 1,
          "height": 106.5,
          "marginBottom": 5,
          "marginLeft": -30,
          "marginTop": 19.61764705882353,
          "width": 632.1428571428571,
        },
        undefined,
      ]
    }
    dropdownTextHighlightStyle={
      Object {
        "color": "#666666",
        "fontFamily": "Quicksand-Medium",
        "fontSize": 25.142857142857142,
      }
    }
    dropdownTextStyle={
      Object {
        "color": "#666666",
        "fontFamily": "Quicksand-Medium",
        "fontSize": 25.142857142857142,
      }
    }
    keyboardShouldPersistTaps="never"
    onDropdownWillHide={[Function]}
    onDropdownWillShow={[Function]}
    options={
      Array [
        0,
        1,
        2,
      ]
    }
    renderRow={[Function]}
    renderSeparator={[Function]}
    scrollEnabled={true}
    showsVerticalScrollIndicator={true}
    style={
      Array [
        Object {
          "borderColor": "#bdbdbd",
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 78.47058823529412,
          "justifyContent": "center",
          "paddingHorizontal": 15,
          "width": 642.8571428571429,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#666666",
                "fontFamily": "Quicksand-Medium",
                "fontSize": 25.142857142857142,
              },
              undefined,
            ]
          }
        />
        <Image
          resizeMode="contain"
          style={
            Object {
              "alignSelf": "flex-end",
              "height": 29.426470588235293,
              "marginBottom": 3,
              "width": 27.857142857142858,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Component snapshot testing renders without data 2`] = `
<View>
  <View
    animated={true}
    defaultIndex={-1}
    defaultValue="Please select..."
    disabled={false}
    dropdownStyle={
      Array [
        Object {
          "borderColor": "#bdbdbd",
          "borderWidth": 1,
          "height": 177.5,
          "marginBottom": 5,
          "marginLeft": -30,
          "marginTop": 19.61764705882353,
          "width": 632.1428571428571,
        },
        undefined,
      ]
    }
    dropdownTextHighlightStyle={
      Object {
        "color": "#666666",
        "fontFamily": "Quicksand-Medium",
        "fontSize": 25.142857142857142,
      }
    }
    dropdownTextStyle={
      Object {
        "color": "#666666",
        "fontFamily": "Quicksand-Medium",
        "fontSize": 25.142857142857142,
      }
    }
    keyboardShouldPersistTaps="never"
    onDropdownWillHide={[Function]}
    onDropdownWillShow={[Function]}
    options={
      Array [
        0,
        1,
        2,
        3,
        4,
      ]
    }
    renderRow={[Function]}
    renderSeparator={[Function]}
    scrollEnabled={true}
    showsVerticalScrollIndicator={true}
    style={
      Array [
        Object {
          "borderColor": "#bdbdbd",
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 78.47058823529412,
          "justifyContent": "center",
          "paddingHorizontal": 15,
          "width": 642.8571428571429,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#666666",
                "fontFamily": "Quicksand-Medium",
                "fontSize": 25.142857142857142,
              },
              undefined,
            ]
          }
        />
        <Image
          resizeMode="contain"
          style={
            Object {
              "alignSelf": "flex-end",
              "height": 29.426470588235293,
              "marginBottom": 3,
              "width": 27.857142857142858,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Component snapshot testing renders without data 3`] = `
<View>
  <View
    animated={true}
    defaultIndex={-1}
    defaultValue="Please select..."
    disabled={false}
    dropdownStyle={
      Array [
        Object {
          "borderColor": "#bdbdbd",
          "borderWidth": 1,
          "height": 177.5,
          "marginBottom": 5,
          "marginLeft": -30,
          "marginTop": 19.61764705882353,
          "width": 632.1428571428571,
        },
        undefined,
      ]
    }
    dropdownTextHighlightStyle={
      Object {
        "color": "#666666",
        "fontFamily": "Quicksand-Medium",
        "fontSize": 25.142857142857142,
      }
    }
    dropdownTextStyle={
      Object {
        "color": "#666666",
        "fontFamily": "Quicksand-Medium",
        "fontSize": 25.142857142857142,
      }
    }
    keyboardShouldPersistTaps="never"
    onDropdownWillHide={[Function]}
    onDropdownWillShow={[Function]}
    options={
      Array [
        0,
        1,
        2,
        3,
        4,
        5,
      ]
    }
    renderRow={[Function]}
    renderSeparator={[Function]}
    scrollEnabled={true}
    showsVerticalScrollIndicator={true}
    style={
      Array [
        Object {
          "borderColor": "#bdbdbd",
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 78.47058823529412,
          "justifyContent": "center",
          "paddingHorizontal": 15,
          "width": 642.8571428571429,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#666666",
                "fontFamily": "Quicksand-Medium",
                "fontSize": 25.142857142857142,
              },
              undefined,
            ]
          }
        />
        <Image
          resizeMode="contain"
          style={
            Object {
              "alignSelf": "flex-end",
              "height": 29.426470588235293,
              "marginBottom": 3,
              "width": 27.857142857142858,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
